File: swift/sqare.swift
import Foundation

func countSquares() {

    guard let nStr = readLine(), let n = Int(nStr) else {
        print("Number count input error")
        return
    }

    guard let numbersStr = readLine() else {
        print("Number imput error")
        return
    }

    let numbers = numbersStr.split(separator: " ").compactMap { Int($0) }

    guard numbers.count == n else {
        print("Number count mismatch")
        return
    }

    var count = 0

    for number in numbers {
        let root = Int(sqrt(Double(number)))
        if root * root == number {
            count += 1
        }
    }
    print(count)
}

countSquares()

File: elixir/sqare.exs
defmodule PerfectSquares do
  def count_perfect_squares do
    n_str = IO.gets("") |> String.trim()
    {n, _} = Integer.parse(n_str)

    numbers_str = IO.gets("") |> String.trim()

    numbers =
      numbers_str
      |> String.split(" ")
      |> Enum.map(&String.to_integer/1)

      if length(numbers) != n do
      IO.puts("Number count mismatch")
      :ok

    else
      count =
        numbers
        |> Enum.count(fn number ->
          root = :math.sqrt(number) |> round()
          root * root == number
        end)

      IO.puts(count)
    end
  end
end

PerfectSquares.count_perfect_squares()
